### ISSUE TITLE
Agents page blows up when elastic agent plugin goes missing

### ISSUE THREAD

**Issue by ketan** [gocd/gocd#3305]
##### Issue Type

- Bug Report

##### Steps to Reproduce

- Add the following XML snippet —

```
  <agents>
    <agent hostname="93a54c58a4c4" ipaddress="172.17.0.2" uuid="411c2c16-7967-4957-80ed-f6af2711e570" elasticAgentId="agentId" elasticPluginId="pluginId" />
    <agent hostname="3d53d9ae59e3" ipaddress="172.17.0.2" uuid="3313f918-4f52-40ab-b9ba-83cdefe2e15e" elasticAgentId="agentId" elasticPluginId="pluginId" />
  </agents>
```

- Goto the agents page.

##### Stacktrace (mangled because of production JS)

```
agents-3aa322d….js:1 Uncaught TypeError: Cannot read property 'imageUrl' of undefined
    at Object.view (agents-3aa322d….js:1)
    at l (vendor-and-helpers.chunk-b3453b6….js:3)
    at r (vendor-and-helpers.chunk-b3453b6….js:3)
    at n (vendor-and-helpers.chunk-b3453b6….js:3)
    at u (vendor-and-helpers.chunk-b3453b6….js:3)
    at r (vendor-and-helpers.chunk-b3453b6….js:3)
    at n (vendor-and-helpers.chunk-b3453b6….js:3)
    at u (vendor-and-helpers.chunk-b3453b6….js:3)
    at r (vendor-and-helpers.chunk-b3453b6….js:3)
    at n (vendor-and-helpers.chunk-b3453b6….js:3)
﻿
​```

**Comment by GaneshSPatil** [comment]
@ketan @rajiesh -- this bug was again introduced as part of [this](https://github.com/gocd/gocd/commit/9b4d79a4a9d1261dae1090e0dbe8483b22d05a41) commit. The commit is merged for current (`v18.2.0`) release. 

Reopening the issue and adding it to the current milestone.

**Comment by rajiesh** [comment]
Verified on 18.2.0 (6211-48d5a54cd6525e54b4692cb757a6ad9896e46392)
